src/Db/Adapter/Pdo/AbstractPdo.php

```diff
@@ -198,7 +198,22 @@ public function commit(bool $nesting = true): bool
              */
             $this->transactionLevel--;
 
-            return $this->pdo->commit();
+            $result = $this->pdo->commit();
+
+            /**
+             * When error mode is set to silent or warning, we need to check result and trigger event only when
```

Comment:
This is the "silent" mode I refer in my comment

Reply:
Yes, it is about `PDO::ATTR_ERRMODE` value that can be one of those three:

1. PDO::ERRMODE_EXCEPTION - we don't need to check anything, execution will stop
2. PDO::ERRMODE_WARNING - warning will be generated and called pdo method will return false
3. PDO::ERRMODE_SILENT - there will be no warning but called pdo method will return false.

So in cases 2 and 3 we just need to check return from the method. We don't need to know which mode it is.
In case error mode is in exception then event will not be triggered anyway.